crates/matrix-sdk-ui/src/timeline/event_item/content/message.rs

```diff
@@ -373,20 +379,69 @@ impl RepliedToEvent {
 
         debug!(event_type = %event.event_type(), "got deserialized event");
 
-        let Some(AnyMessageLikeEventContent::RoomMessage(c)) = event.original_content() else {
-            warn!("can't get details, event is redacted or not a RoomMessage");
-            return Err(TimelineError::UnsupportedEvent);
-        };
+        let content = match event.original_content() {
+            Some(content) => match content {
+                AnyMessageLikeEventContent::RoomMessage(c) => {
+                    // Assume we're not interested in reactions in this context.
+                    let reactions = ReactionsByKeyBySender::default();
+
+                    TimelineItemContent::Message(Message::from_event(
+                        c,
+                        extract_room_msg_edit_content(event.relations()),
+                        &vector![],
+                        reactions,
+                    ))
+                }
+
+                AnyMessageLikeEventContent::Sticker(content) => {
+                    // Assume we're not interested in reactions in this context.
+                    let reactions = ReactionsByKeyBySender::default();
+                    TimelineItemContent::Sticker(Sticker { content, reactions })
+                }
 
-        // Assume we're not interested in reactions in this context.
-        let reactions = Default::default();
+                AnyMessageLikeEventContent::RoomEncrypted(content) => {
+                    let utd_info = as_variant!(
+                        &timeline_event.kind,
+                        TimelineEventKind::UnableToDecrypt { utd_info, .. } => utd_info
+                    );
+
+                    let utd_cause = if let Some(utd_info) = utd_info {
+                        UtdCause::determine(
+                            timeline_event.raw(),
+                            room_data_provider.crypto_context_info().await,
+                            &utd_info,
+                        )
+                    } else {
+                        UtdCause::Unknown
+                    };
```

Comment:
Seems simpler to use a single `match` for this, no?

Reply:
Good idea!